### ISSUE TITLE
[Feature Request] PIN or Top Most icons

### ISSUE THREAD

**Issue by neoOpus** [fwextensions/QuicKey#98]
Hi, I would like to suggest a new feature, which consist of a bar under the search where it is possible to pin tabs and see them as just icons (inside tiny rounded-box icon) and maybe have a feature to fill it will the top visited tabs (or those who have that name)... this will allow even faster switching without having to type anything or scrolling.

**Comment by fwextensions** [comment]
Interesting.  Are you thinking you'd mostly be clicking these icons with the mouse?  Would the pinned tabs in this toolbar also include pinned browser tabs?  Or be a different set?

If this was added, I think I'd put it above the search box, so you could use up-arrow to navigate to it, then left/right to select different icons.  

One concern I'd have is that it could look somewhat duplicative to the recent list, since as soon as you clicked a pinned tab, it would appear at the top of the recent list.  Also, it's not clear what the right order for those would be.  The order in which they were added seems arbitrary.  In recent order?  Then they tend to duplicate the recent list.  Maybe they need to be manually reordered, though that drag and drop gets a little tricky.

**Comment by neoOpus** [comment]
I wrote an answer with a lot of details yesterday, but it is lost :/

I will come back later to write it again.

**Comment by neoOpus** [comment]
> Interesting. Are you thinking you'd mostly be clicking these icons with the mouse? Would the pinned tabs in this toolbar also include pinned browser tabs? Or be a different set?

Thank you for taking my idea into consideration, actually I am more a mouse user than keyboard and I always try to have most of the interaction few clicks away with the less distraction possible. 

I previously wrote a full detail about the tiny tools and extensions that I use on daily basis to give you an idea, and also to share them as they increase productivity and focus tremendously, but I will stick to just the request on this one unless you ask me to.

Yes, the buttons should be placed somewhere on top, just the favicons of the websites as the text + urls is quite overwhelming sometimes for fast interaction with zero distraction.

Pinned tabs are easy to find unless they are in another window then it shouldn't be necessary to list them there (unless there are many windows with different pinned tabs on them)

The icons should auto-organize by order of interaction (maybe view)

**Comment by neoOpus** [comment]
> If this was added, I think I'd put it above the search box, so you could use up-arrow to navigate to it, then left/right to select different icons.

You are right and I have no doubt that you will find the best way to make this functional without interfering with what some users are already used to.

**Comment by neoOpus** [comment]
> One concern I'd have is that it could look somewhat duplicative to the recent list, since as soon as you clicked a pinned tab, it would appear at the top of the recent list. Also, it's not clear what the right order for those would be. The order in which they were added seems arbitrary. In recent order?

Maybe having the possibility, the pin and re-arrange the icons is the best option, so they will be some other pinned (but that will not go out of groups like the browser default pins, they will not move the tabs from their position they will act just like a shortcut to them, and they can be unpinned from that bar... it is maybe implant but that would be a great way to move between hundreds of tabs who are in different tab groups and windows... that augment the browsers default implantation with isn't a good one since grouping isn't really representative of how tabs should be presented if they are grouped hierarchically then when you move the tab to another group you broke that relationship and many times it is required to do so, if they are grouped by domain then they don't have that connection of the subject (theme)... so until they find a better way to group and find them this would be the best way to patch that gap.

I hope it makes sense.

**Comment by fwextensions** [comment]
> I wrote an answer with a lot of details yesterday, but it is lost :/

Bummer!  You can always save the comment and then edit it to add more, so you can save bits as you go.

**Comment by neoOpus** [comment]
> > I wrote an answer with a lot of details yesterday, but it is lost :/
> 
> Bummer! You can always save the comment and then edit it to add more, so you can save bits as you go.

Yeah but I didn't want you to read a draft, I thought that there is a sort of autosave option since sometimes when I refresh the page what I wrote is still there (I don't know if it due to GH or a userscript)
